src/main/java/duke/AddCommand.java

```diff
@@ -0,0 +1,175 @@
+           break;
+        }
+    }
+
+    /**
+     * Returns user input.
+     *
+     * @return User input as a String.
+     */
+    public String getUserInput() {
+        return this.userInput;
+    }
+
+    /**
+     * Adds a ToDo task.
+     *
+     * @param tasks Task List containing tasks.
+     * @param storage Writes the task being added into file.
+     * @param ui Handles printing of user interaction.
+     * @param userInput User input as a String.
+     * @throws DukeException When description of a ToDo task is empty.
+     * @throws IOException When writing to file fails.
+     */
+    public void addToDo(TaskList tasks, Storage storage, Ui ui,
+                        String userInput) throws DukeException, IOException {
+        if (!userInput.substring(4).isBlank()) { //if got space behind, it will add also
+            ToDo todo = new ToDo(userInput.substring(5));
+            tasks.addTask(todo); //adds into tasks list
+            ui.printAddTodo(todo, tasks);
+            storage.writeToFile(tasks.getTasks());
+        } else {
+            System.out.println(Ui.getLine());
+            System.out.println(Ui.getBot());
+            throw new DukeException("The description of todo cannot be empty!");
+        }
+    }
+
+    /**
+     * Adds a Deadline Task.
+     *
+     * @param tasks Task List containing tasks.
+     * @param storage Writes the task being added into file.
+     * @param ui Handles printing of user interaction.
+     * @param userInput User input as a String.
+     * @throws DukeException When input for Deadline is invalid, respective error messages are printed.
+     */
+    public void addDeadline(TaskList tasks, Storage storage, Ui ui, String userInput)
+            throws DukeException {
+        String[] input = userInput.split(" ");
+        if (!userInput.substring(8).isBlank()) {
+            try {
+                String de = String.join(" ",
+                        Arrays.copyOfRange(input, 1, input.length));
+                String description = de.split(" /by ")[0];
+                String date = de.split(" /by ")[1];
+                Deadline deadline = new Deadline(description, date);
+                tasks.addTask(deadline);
+                ui.printAddDeadline(deadline, tasks);
+                storage.writeToFile(tasks.getTasks());
+            } catch (PatternSyntaxException | ArrayIndexOutOfBoundsException ex) {
+                System.out.println(Ui.getLine());
+                System.out.println(Ui.getBot());
+                throw new DukeException(
+                        "You have keyed in an invalid input for 'deadline'!");
+            } catch (DateTimeParseException | IOException ex) {
+                System.out.println(Ui.getLine());
+                System.out.println(Ui.getBot());
+                throw new DukeException(
+                        "Please key in your deadline in the form:\n" +
+                                " /by <dd/MM/yyyy hh:mm AM/PM>");
```

Comment:
I like how there are line breaks to improve the readability of the error message.